supernode/daemon/mgr/task/manager.go

```diff
@@ -272,6 +272,15 @@ func (tm *Manager) UpdatePieceStatus(ctx context.Context, taskID, pieceRange str
 			pieceRange, taskID, pieceUpdateRequest.ClientID)
 	}
 
+	// when a peer success to download a piece from supernode,
+	// and the load of supernode for the taskID should be decremented by one.
+	if tm.cfg.IsSuperPID(pieceUpdateRequest.DstPID) {
+		_, err := tm.progressMgr.UpdateSuperLoad(ctx, taskID, -1, -1)
```

Comment:
The load of supernode cannot be decreased here because that the `limit` is less than zero and the `if` statement below would not execute the code `loadState.loadValue.Add(delta)`.

```go
	if limit > 0 && loadState.loadValue.Add(delta) > limit {
		loadState.loadValue.Add(-delta)
		return false, nil
	}
```

Reply:
Thanks for pointing that. And I have updated it. PTAL again. THX>